index.bs

```diff
@@ -932,6 +932,159 @@ requests to start a WebDriver session, it must:
 
 </div>
 
+# Sandboxed Script Execution # {#sandbox}
+
+A common requirements for automation tools is to execute scripts which have
+access to the DOM of a document, but don't have information about any changes to
+the DOM APIs made by scripts running in the browsing context containing the
+document.
+
+A [=BiDi session=] has a <dfn>context sandbox map</dfn> which is a weak map
+in which the keys are [=browsing contexts=] and the values are weak maps between
+strings and {{SandboxWindowProxy}} objects.
+
+Note: The definition of sandboxes here is an attempt to codify the behaviour of
+existing implementations. It exposes parts of the implementations that have
+previously been considered internal by specifications, in particular the
+distinction between the internal state of platform objects (which is typically
+implemented as native objects in the main implementation language of the browser
+engine) and the ECMAScript-visible state. Because existing sandbox
+implementations happen at a low level in the engine, implementations converging
+toward the specficication in all details might be a slow process. In the
+meantime, implementors are encouraged to provide detailed documentation on any
+differences with the specification, and users of this feature are encouraged to
+explicitly test that scripts running in sandboxes work in all implementations.
+
+## Sandbox Realms ## {#sandbox-realm}
+
+Each sandbox is a unique EMCAScript [=Realm=]. However the sandbox realm
+provides access to platform objects in an existing {{Window}} realm via
+{{SandboxProxy}} objects.
+
+<div algorithm>
+To <dfn>get or create a sandbox realm</dfn> given |name| and |source browsing context|:
+
+1. If [=context sandbox map=] does not contain |name|, set |source browsing
+   context|[|name|] to a new map.
+
+1. Let |sandboxes| be [=context sandbox map=][|name|].
+
+1. If |sandboxes| does not contain |name|, set |sandboxes|[|name|] to [=create
+   a sandbox realm=] with |source browsing context|.
+
+1. Return |sandboxes|[|name|].
+
+</div>
+
+<div algorithm>
+To <dfn>create a sandbox realm</dfn> with |source browsing context|:
+
+1. Let |agent| be
+
+1. Let |realm| be the result of [=creating a new JavaScript realm=] given |agent| and the following customizations:
+
+  * For the global object, create a new {{SandboxWindowProxy}} whose
+    \[[Wrapped]] internal slot is set to |source browsing context|'s {{WindowProxy}}
+
+1. Let |environment settings| be an [=environment settings object=] whose
+   algorithms are defined as follows:
+
+   <dl>
+       <dt>The [=realm execution context=]
+       <dd>
+
+       <dt>The [=module map=]
+       <dd>Issue: Not sure if this ought to be the module map of the assocaited
```

Comment:
I suspect this should be an empty module map. If I'm understanding correctly, using the document's module map would mean that any names exported by module scripts on the page are now accessible in the sandbox if it imports the modules. This would break the isolation between the two realms.